### PR TITLE
update footer - dummy look for social media icons

### PR DESCRIPTION
**Pull request by bbrcknl**
Signed-off-by: Birgit Brecknell <birgit@sel4.systems>

### CODE REVIEW

**Comment by bbrcknl** [comment]
Hi @lsf37 
I've started playing with the footer. It's still incredibly ugly (e.g. the white background of the social media icons stand out from the greyish footer background). Also the alignment is terrible.
Having said that, is this the sort of thing you were thinking of for the footer?

**Comment by lsf37** [comment]
Cool, yes, that is going in the right direction, also makes the icons a bit more prominent.

If we can get the background transparent and move the icons a bit more to the right, it looks pretty good to me.

Now for the hard part: if we do this design, the footer should always be at least be at the bottom of the browser window or further down (for larger pages). I'm sure there is some css magic that can make that happen, but it's probably not pretty.

**Comment by lsf37** [comment]
Oh, and we probably need to check this on mobile/small screens (haven't done that yet)

**Comment by gernotheiser** [comment]
How about putting the icons into a box next to (rather than below) the footer text?

**Comment by lsf37** [comment]
> How about putting the icons into a box next to (rather than below) the footer text?

Right, that is the screen size issue: they are next to the footer text for me and that looks much better. I think that should still work for slightly smaller screen sizes (and I agree we should do that for those). It'll not work on actual mobile sizes any more. For those I'd propose to put the icons on top and the copyright stuff underneath with a bit of space.

**Comment by gernotheiser** [comment]
Somehow the LinkedIn logo comes out really huge on my preview

**Comment by bbrcknl** [comment]
@gernotheiser That may be due to a cashed image from when I was playing around with it earlier. The linked in logo should be the same height as the others now.

**Comment by gernotheiser** [comment]
Shift-Reload in Chrome shows the same. I just tried loading it through ToR (which I never used on that page) and it's still the same

**Comment by gernotheiser** [comment]
this looks good on both large screen and phone

**Comment by bbrcknl** [comment]
@gernotheiser @lsf37 
- footer now sticks to bottom of page
- different footer layout for mobiles and bigger screens
- svg logos instead of png
- Please check; I'll create a final PR next week, after I fix some the validator issues

**Comment by lsf37** [comment]
Cool, it's almost there. The horizontal layout for small/big screens is very nice, but the footer should not always be visible at the bottom, only when it fits, i.e. it shouldn't obscure other text, esp on small screens.

Basically `fixed` is not the right CSS attribute for this. Instead, you want the page to be at least screen-height long, but allowed to be longer if needed, and the footer always underneath the content. Does this make sense?

**Comment by bbrcknl** [comment]
@lsf37 I've removed the sticky footer. It is now back to being at the bottom of the page. :-)
The PR should be close to done. There's a link checker error, but this is related to a summit webpage, not my work.

**Comment by lsf37** [comment]
> @lsf37 I've removed the sticky footer. It is now back to being at the bottom of the page. :-)

It's not sticky any more, but it's not actually at the bottom :-). What I'm after is stretchable space underneath the main text, so that the footer is at the bottom of the browser window (like in fixed) when the text is too short to fill the browser window, but not obscuring the text (unlike fixed) when the text takes up the browser window.

(For browser windows that wouldn't be so important, but you don't want to take up half the screen with a footer on mobile with fixed, and you don't want to have the footer in the middle of the screen on a big screen)

**Comment by bbrcknl** [comment]
@lsf37 I see what you mean - I'll fix it now.

**Comment by lsf37** [comment]
Awesome, this is great now!

**Comment by bbrcknl** [comment]
Almost there... just fixing up the footer height for mobiles...


---

Dr. Birgit Brecknell

Project Officer

Trustworthy Systems, UNSW, ***@***.******@***.***>

seL4 Foundation, ***@***.******@***.***>

0433 880 571



Mon 9am-5pm

Wed 9am-12pm

Fri 9am-5pm

________________________________

**Comment by bbrcknl** [comment]
@lsf37 Ready for merge now :-)

File: content/images/icons/facebook.svg.licence [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Copyright 2022 Facebook
+See https://www.facebook.com/brand/resources/facebookapp/guidelines for brand guidelines
```
Comment (lsf37) [inline content/images/icons/facebook.svg.licence:2]: Not sure why the license check is passing for this one -- it shouldn't, we need an SPDX license reference in here. I'll have a look.

Comment (bbrcknl) [reply content/images/icons/facebook.svg.licence:2]: Ok, thanks. I discussed this with Peter, and he told me to set up a file with a link to the FB branding page. I looked for licences for all the social media icons and came up blank.

Comment (lsf37) [reply content/images/icons/facebook.svg.licence:2]: Hm, yes there is a problem in the `dep5` file. It says:

```
Files: content/images/icons/* content/robots.txt content/favicon.ico
Copyright: 2020 seL4 Project a Series of LF Projects, LLC.
License: CC-BY-SA-4.0
```

Apparently we had only our own icons in there before. I'll try to make that list more precise so it doesn't declare additional icons as Creative Commons.

This still leaves us with the problem that the actual license for these new icons needs to be in the LICENSES file in the repository. A link is not sufficient.

Comment (lsf37) [reply content/images/icons/facebook.svg.licence:2]: Ok, this should work:
```suggestion
Copyright 2022 Facebook
SPDX-License-Identifier: LicenseRef-UsedWithPermission

See https://www.facebook.com/brand/resources/facebookapp/guidelines for brand guidelines
```

File: content/images/icons/linkedin.svg.licence [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Copyright 2022 LinkedIn
+See https://brand.linkedin.com/downloads for brand guidelines
```
Comment (lsf37) [inline content/images/icons/linkedin.svg.licence:2]:
```suggestion
Copyright 2022 LinkedIn
SPDX-License-Identifier: LicenseRef-UsedWithPermission

See https://brand.linkedin.com/downloads for brand guidelines
```

File: content/images/icons/twitter.svg.licence [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Copyright 2022 Twitter
+See https://about.twitter.com/en/who-we-are/brand-toolkit for brand guidelines
```
Comment (lsf37) [inline content/images/icons/twitter.svg.licence:2]:
```suggestion
Copyright 2022 Twitter
SPDX-License-Identifier: LicenseRef-UsedWithPermission

See https://about.twitter.com/en/who-we-are/brand-toolkit for brand guidelines
```

File: content/images/icons/youtube.svg.licence [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Copyright 2022 Youtube
+See https://www.youtube.com/howyoutubeworks/resources/brand-resources/ for brand guidelines
```
Comment (lsf37) [inline content/images/icons/youtube.svg.licence:2]:
```suggestion
Copyright 2022 Youtube
SPDX-License-Identifier: LicenseRef-UsedWithPermission

See https://www.youtube.com/howyoutubeworks/resources/brand-resources/ for brand guidelines
```

**Review by lsf37 — APPROVED** [review]
Very nice. If you adjust the license files, it's ready to merge.

**Comment by bbrcknl** [comment]
Do you know where I can find these licence files? Peter and I looked and couldn't find licences for social media icons. All we found were pages with branding info. These pages had downloadable icons/logos, but no licences in the zip files. This is why we created licences that pointed to the branding pages.


---

Dr. Birgit Brecknell

Project Officer

Trustworthy Systems, UNSW, ***@***.******@***.***>

seL4 Foundation, ***@***.******@***.***>

0433 880 571



Mon 9am-5pm

Wed 9am-12pm

Fri 9am-5pm

________________________________

**Comment by lsf37** [comment]
> Do you know where I can find these licence files? Peter and I looked and couldn't find licences for social media icons. All we found were pages with branding info. These pages had downloadable icons/logos, but no licences in the zip files. This is why we created licences that pointed to the branding pages.
> […](#)

Yes, basically they don't really give you a license, so this is a bit annoying. They do give you permission to use the icons on a website in accordance with those guidelines, but they can change those at any time etc. 

For our purposes I think it's Ok if we have an SPDX identifier that expresses that, so I've used `LicenseRef-UsedWithPermission` for it in the code suggestions above. Keeping the link in addition is good for our own information.

**Comment by bbrcknl** [comment]
I see! (finally) So much to learn. Thanks for fixing up the licence files.

**Comment by lsf37** [comment]
I don't think you included those code suggestions :-) (have to press "commit" first).

No worries, I'll add them to #190 instead.

**Comment by bbrcknl** [comment]
Oh :-( 
Thanks in advance for adding them to 190

**Comment by lsf37** [comment]
No worries!

**Pull request merged**